Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
 import Mathlib.Analysis.Calculus.ParametricIntegral
 import Mathlib.Analysis.Fourier.AddCircle
 import Mathlib.Analysis.Fourier.FourierTransform
+
 /-!
 # Derivative of the Fourier transform
 
-In this file we compute the Fréchet derivative of `𝓕 f`, where `f` is a function such that both
-`f` and `v ↦ ‖v‖ * ‖f v‖` are integrable. Here `𝓕` is understood as an operator `(V → E) → (W → E)`,
-where `V` and `W` are normed `ℝ`-vector spaces and the Fourier transform is taken with respect to a
-continuous `ℝ`-bilinear pairing `L : V × W → ℝ`.
+In this file we compute the Fréchet derivative of the Fourier transform of `f`, where `f` is a
+function such that both `f` and `v ↦ ‖v‖ * ‖f v‖` are integrable. Here the Fourier transform is
+understood as an operator `(V → E) → (W → E)`, where `V` and `W` are normed `ℝ`-vector spaces
+and the Fourier transform is taken with respect to a continuous `ℝ`-bilinear
+pairing `L : V × W → ℝ` and a given reference measure `μ`.
+
+We give specialized versions of these results on inner product spaces (where `L` is the scalar
+product) and on the real line, where we express the one-dimensional derivative in more concrete
+terms, as the Fourier transform of `x * f x`.
+
+## Main definitions and results
+
+We introduce one handy definition:
+
+* `VectorFourier.fourierSMulRight L f`: given `f : V → E` and `L` a bilinear pairing
+  between `V` and `W`, then this is the function `fun v ↦ -(2 * π * I) (L v ⬝) • f v`,
+  from `V` to `Hom (W, E)`.
+  This is essentially `ContinousLinearMap.smulRight`, up to the factor `- 2πI` designed to make sure
+  that the fourier integral of `fourierSMulRight L f` is the derivative of the Fourier
+  integral of `f`.
 
-We also give a separate lemma for the most common case when `V = W = ℝ` and `L` is the obvious
-multiplication map.
+With this definition, the statements read as follows, first in a general context
+(arbitrary `L` and `μ`):
+
+* `VectorFourier.hasFDerivAt_fourierIntegral`: the Fourier integral of `f` is differentiable, with
+    derivative the Fourier integral of `fourierSMulRight L f`.
+* `VectorFourier.differentiable_fourierIntegral`: the Fourier integral of `f` is differentiable.
+* `VectorFourier.fderiv_fourierIntegral`: formula for the derivative of the Fourier integral of `f`.
+
+These statements are then specialized to the case of the usual Fourier transform on
+finite-dimensional inner product spaces with their canonical Lebesgue measure (covering in
+particular the case of the real line), replacing the namespace `VectorFourier` by
+the namespace `Real` in the above statements.
+
+We also give specialized versions of the one-dimensional real derivative
+in `Real.deriv_fourierIntegral`.
 -/
 
 noncomputable section
 
 open Real Complex MeasureTheory Filter TopologicalSpace
 
-open scoped FourierTransform Topology
+open scoped FourierTransform Topology BigOperators
+
+attribute [local instance 2000] secondCountableTopologyEither_of_left
```

Comment:
This priority is fairly opaque to me: is there an explanation that could fit in a comment?